packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -738,27 +711,27 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
     function _computePendingStakes(address voterAddress, uint256 amount) internal override {
         if (_inActiveReveal()) {
             uint256 currentRoundId = getCurrentRoundId();
-            // Now freeze the round variables as we do not want the cumulativeActiveStakeAtRound to change based on the
-            // stakes during the active reveal phase. This only happens if the first action within the active reveal is
-            // someone staking, rather than someone revealing their vote.
+            // Freeze round variables to prevent cumulativeActiveStakeAtRound from changing based on the stakes during
+            // the active reveal phase. This will happen if the first action within the reveal is someone staking.
             _freezeRoundVariables(currentRoundId);
-            // Finally increment the pending stake for the voter by the amount to stake. Together with the omission of
-            // the new stakes from the cumulativeActiveStakeAtRound for this round, this ensures that the pending stakes
-            // of any voter are not included in the slashing calculation for this round.
-            _setPendingStake(voterAddress, currentRoundId, amount);
+            // Increment pending stake for voter by amount. With the omission of stake from cumulativeActiveStakeAtRound
+            // for this round, ensure that the pending stakes is not included in the slashing calculation for this round.
+            _incrementPendingStake(voterAddress, currentRoundId, amount);
         }
     }
 
     // Updates the slashing trackers of a given account based on previous voting activity. This traverses all resolved
     // requests for each voter and for each request checks if the voter voted correctly or not. Based on the voters
     // voting activity the voters balance is updated accordingly. The caller can provide a maxTraversals parameter to
-    // limit the number of resolved requests to traverse in this call. This is useful if the number of resolved requests
-    // is large and the update needs to be split over multiple transactions.
+    // limit the number of resolved requests to traverse in this call to bound the gas used. Note that throughout this
+    //function we use the voters unappliedSlash to track any unapplied slashing still to be applied to the voter.
     function _updateAccountSlashingTrackers(address voterAddress, uint64 maxTraversals) internal {
         VoterStake storage voterStake = voterStakes[voterAddress];
-        int256 slash = voterStake.unappliedSlash; // Load in any unapplied slashing from the previous iteration.
```

Comment:
we can rather re-use the `unappliedSlash` variable throughout the function. this saves a slot and decreases an extra check at the end.